src/FileIo.cpp

```diff
@@ -167,6 +167,82 @@ QString FileIo::readFile(QString const& _url)
 	}
 }
 
+int FileIo::getFileSize(QString const& _url)
+{
+	try
+	{
+		QString path = pathFromUrl(_url);
+		QFile file(path);
+		return file.size();
+	}
+	catch (...)
+	{
+		manageException();
+	}
+
+	return 0;
+}
+
+bool FileIo::isFileText(QString const& _url)
+{
+	try
+	{
+		QString path = pathFromUrl(_url);
+		QFile file(path);
+		if (file.open(QIODevice::ReadOnly | QIODevice::Text))
+		{
+			QByteArray buf = file.read(file.bytesAvailable());
+
+			if (buf.size() >= 2)
+			{
+				//FF FE        UTF - 16, little endian
+				if ((quint8)buf[0] == 0xFF && (quint8)buf[1] == 0xFE)
+					return true;
+
+				//FE FF        UTF - 16, big endian
+				if ((quint8)buf[0] == 0xFE && (quint8)buf[1] == 0xFF)
+					return true;
+			}
+
+			if (buf.size() >= 3)
+			{
+				//EF BB BF     UTF - 8
+				if ((quint8)buf[0] == 0xEF && (quint8)buf[1] == 0xBB && (quint8)buf[2] == 0xBF)
+					return true;
+			}
+
+			if (buf.size() >= 4)
+			{
+				//FF FE 00 00  UTF - 32, little endian
+				if ((quint8)buf[0] == 0xFF && (quint8)buf[1] == 0xFE && (quint8)buf[2] == 0x00 && (quint8)buf[3] == 0x00)
+					return true;
+
+				//00 00 FE FF  UTF - 32, big - endian
```

Comment:
Does this check really work? Do we read files in utf-32 big endian if they start with this sequence?